### PR TITLE
Use new webView data source in prod

### PR DESCRIPTION
**Pull request by nelsitoPuglisi**
## Goal

Track web view urls as span events of the session span instead of breadcrumbs.

## Testing

Unit test, Integration test and manual test.

### CODE REVIEW

**Comment by nelsitoPuglisi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/680?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#680** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/680?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#674** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/674?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nelsitoPuglisi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt [hunk 1]
```diff
@@ -36,7 +36,8 @@ internal class EmbraceBreadcrumbService(
 
     private val breadcrumbDataSource =
         BreadcrumbDataSource(configService.breadcrumbBehavior, sessionSpanWriter, logger)
-    private val legacyWebViewBreadcrumbDataSource = LegacyWebViewBreadcrumbDataSource(configService, logger)
+    private val webViewBreadcrumbDataSource =
+        WebViewUrlDataSource(configService.breadcrumbBehavior, sessionSpanWriter, logger)
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt:40]: This needs to be defined in `DataSourceModule` to get picked up in the payload

**Review by fractalwrench — COMMENTED** [review]
Looks good barring one inline comment. I'd also like to confirm that this has been tested E2E before this gets merged.

It would be good to write an integration test too that checks the feature.

**Comment by nelsitoPuglisi** [comment]
> Looks good barring one inline comment. I'd also like to confirm that this has been tested E2E before this gets merged.
> 
> It would be good to write an integration test too that checks the feature.

Yes, I tested it manually and checked the json received in the backend, and the dashboard displays it properly. 

Besides that, this integration test covers the funcionality https://github.com/embrace-io/embrace-android-sdk/pull/680/files#diff-49417e80327f621fcf052fcad356035ea38339cb3faba2de6bc3242ae5c81e25R120

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt:40]: A new instance is also defined here though?

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/IntegrationTestRule.kt [hunk 2]
```diff
@@ -212,7 +212,11 @@ internal class IntegrationTestRule(
                 deliveryService = FakeDeliveryService(),
             ),
         val startImmediately: Boolean = true
-    )
+    ) {
+        fun logWebView(url: String) {
+            Embrace.getImpl().logWebView(url)
```
Comment (bidetofevil) [inline embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/IntegrationTestRule.kt:217]: Why do we need to add this here? Shouldn't the integration test be testing the API by going through Embrace.java in the test?

Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/IntegrationTestRule.kt:217]: There is no public api to track web view URL, it's captured through the swazzler hook.

**Pull request merged**